### PR TITLE
Add support for Homebrew formula deps/conflicts

### PR DESCRIPTION
**Pull request by josephfrazier**
Closes https://github.com/OctoLinker/browser-extension/issues/72

Both `depends_on` and `conflicts_with` syntaxes are supported. Here are
some links you can test with:

* `depends_on`: https://github.com/Homebrew/homebrew-core/blob/8a8ac6b7d88ef75fe94adda49c0f4d9df9ef0bfa/Formula/sngrep.rb#L14-16
* `conflicts_with`: https://github.com/Homebrew/homebrew-core/blob/8a8ac6b7d88ef75fe94adda49c0f4d9df9ef0bfa/Formula/osxutils.rb#L15-L18

Here's the syntax documentation:

* `depends_on`: http://www.rubydoc.info/github/Homebrew/brew/master/Formula#depends_on-class_method
* `conflicts_with`: http://www.rubydoc.info/github/Homebrew/brew/master/Formula#conflicts_with-class_method

### CODE REVIEW

**Comment by stefanbuck** [comment]
Amazing, will have a closer look asap. I'm very excited to review it 😉  Are you interested in sharing your first experiences with the project? If so, please open a separate issue for that. Especially I want to know how easy/hard it was to create this PR and if you missed anything. That would be super helpful. 

Thank you so much for your contribution 👍

File: lib/plugins/homebrew-manifest/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    // various people have created. Here are a couple of searches that
+    // illustrate where "depends_on" and "conflicts_with" appear in Ruby files:
+    // * https://github.com/search?l=ruby&langOverride=&q=depends_on&repo=&start_value=1&type=Code
+    // * https://github.com/search?l=ruby&langOverride=&q=conflicts_with&repo=&start_value=1&type=Code
+    return preset('Ruby');
+  }
+
+  parseBlob(blob) {
+    insertLink(blob.el, HOMEBREW, {
+      resolver: 'homebrewFile',
```
Comment (stefanbuck) [inline lib/plugins/homebrew-manifest/index.js]: Please reuse the `relativeFile` resolver for that. I made same changes in the data-attr handling which allows you to reuse this resolver https://github.com/OctoLinker/browser-extension/pull/83

File: lib/plugins/homebrew-manifest/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      path: dirname(blob.path),
```
Comment (stefanbuck) [inline lib/plugins/homebrew-manifest/index.js]: The `relativeFile` resolver need the raw path.

File: lib/plugins/homebrew-manifest/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      target: '$1',
```
Comment (stefanbuck) [inline lib/plugins/homebrew-manifest/index.js]: In according to the PR `target: '$1.rb'` 😉

File: lib/plugins/homebrew-manifest/index.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      type: 'homebrew',
```
Comment (stefanbuck) [inline lib/plugins/homebrew-manifest/index.js]: `type` is not needed

File: lib/resolver/homebrew-file.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { join } from 'path';
+
+export default function ({ path, target }) {
+  return 'https://github.com' + join(path, target + '.rb');
+}
```
Comment (stefanbuck) [inline lib/resolver/homebrew-file.js]: No need anymore for this resolver

**Comment by stefanbuck** [comment]
A couple of comments around the `insertLink` options. Besides that, great job 👍

**Comment by josephfrazier** [comment]
Thanks for the detailed feedback and `relativeFile` accommodations! I've added a couple of commits with your suggested changes. You should be able to [squash them at merge-time](https://help.github.com/articles/about-pull-request-merge-squashing/), if you'd like. Let me know if there's anything else I should do before this can land.

As for the development process, I think I did find a couple of potential areas for improvement. I'll see if I can put a list together in a separate issue. Cheers!

**Comment by stefanbuck** [comment]
Merged 😃 Middle next week I will release a new version. I'm looking forward for your feedback, because I'm really interested in improving the process. 

Thanks again for your contribution. Have a great weekend.

**Pull request merged**